apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GrpcLogConsumer.cs

```diff
@@ -43,7 +45,18 @@ public void Receive(IEnumerable<LogEntry> logs)
             {
                 return;
             }
-            _client.WriteLogEntriesAsync(null, null, EmptyDictionary, logs);
+            _client.WriteLogEntries(null, null, EmptyDictionary, logs);
+        }
+
+        /// <inheritdoc />
+        public async Task ReceiveAsync(
+            IEnumerable<LogEntry> logs, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (!logs.Any())
+            {
+                return;
+            }
+            await _client.WriteLogEntriesAsync(null, null, EmptyDictionary, logs, cancellationToken);
```

Comment:
As per Mike's comments later, this doesn't really need to be async - you can just return _client.WriteLogEntriesAsync.
It may be worth validating that logs is non-null first though.

(Changing from async to non-async will affect how exceptions are propagated, but in this case it would be an argument exception, which is reasonable to be eager IMO.)